node.js

```diff
@@ -400,64 +493,67 @@ async function join(known_node) {
         predecessor = _self;
     }
 
-    // TODO move keys: (predecessor, n]; i.e., (predecessor, _self]
+    // TODO migrate keys: (predecessor, n]; i.e., (predecessor, _self]
+    await migrate_keys();
+
+    // initialize successor table - deviates from SIGCOMM
+    SuccessorTable[0] = FingerTable[0].successor;
 
-    /* vvvvv     vvvvv     TBD debugging code     vvvvv     vvvvv */
     if (DEBUGGING_LOCAL) {
         console.log(">>>>>     join          ");
-        console.log("The FingerTable[] leaving join() is:\n", FingerTable);
-        console.log("The predecessor leaving join() is ", predecessor);
+        console.log("The FingerTable[] leaving {", _self.id, "}.join(", known_node.id, ") is:\n", FingerTable);
+        console.log("The {", _self.id, "}.predecessor leaving join() is ", predecessor);
         console.log("          join     <<<<<\n");
     }
-    /* ^^^^^     ^^^^^     TBD debugging code     ^^^^^     ^^^^^ */
 }
 
+/**
+ * Determine whether a node exists by pinging it.
+ * 
+ * @version 20191103
+ */
 function confirm_exist(known_node) {
-    /**
-     * Determine whether a node exists by pinging it.
-     */
-  // TODO: confirm_exist actually needs to ping the endpoint to ensure it's real
-  return !(_self.id == known_node.id);
+    // TODO: confirm_exist actually needs to ping the endpoint to ensure it's real
```

Comment:
Please open an issue for this and delete the comment

Reply:
Done.